src/screens/SendwyreInput/SendwyreInput.js

```diff
@@ -0,0 +1,288 @@
+  sourceOptionsSelector,
+  destOptionsSelector,
+  defaultSourceSelector,
+  defaultDestSelector,
+} from 'selectors/fiatToCrypto';
+
+// models, types
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+import type { Theme } from 'models/Theme';
+import type { Option } from 'models/Selector';
+import type { SendwyreTrxValues } from 'models/FiatToCryptoProviders';
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  loadAssets: () => void,
+  currencyPairs: [string, string][],
+  loadCurrencyPairs: () => void,
+  sourceOptions: Option[],
+  destOptions: Option[],
+  defaultSource: null | Option,
+  defaultDest: null | Option,
+  countrySupport: $Values<typeof SENDWYRE_SUPPORT>,
+  loadCountrySupport: () => void,
+  theme: Theme,
+};
+
+type State = {
+  sourceCurrency: null | Option,
+  sourceAmount: string,
+  destCurrency: null | Option,
+  isHandlingSubmit: boolean,
+};
+
+const FormWrapper = styled.View`
+  padding: ${spacing.large}px ${spacing.layoutSides}px;
+`;
+
+const ButtonWrapper = styled.View`
+  margin-top: ${spacing.large}px;
+  padding: ${spacing.layoutSides}px;
+`;
+
+class SendwyreInputScreen extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      sourceCurrency: props.defaultSource,
+      sourceAmount: '',
+      destCurrency: props.defaultDest,
+      isHandlingSubmit: false,
+    };
+  }
+
+  componentDidMount() {
+    this.props.loadAssets();
+    this.props.loadCurrencyPairs();
+    this.props.loadCountrySupport();
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    this.tryFillDefaultCurrencies(prevState);
+  }
+
+  onSourceChange = (value: { selector: Option, input: string }) => {
+    this.setState({
+      sourceCurrency: value.selector,
+      sourceAmount: value.input,
+    });
+  }
+
+  onDestChange = (value: { selector: Option }) => {
+    this.setState({ destCurrency: value.selector });
+  }
+
+  onSubmit = async () => {
+    this.setState({ isHandlingSubmit: true });
+
+    const onSubmitCallback: (values: SendwyreTrxValues) => void =
+      this.props.navigation.getParam('onSubmit', () => {});
+
+    const [sourceCurrency, destCurrency] = this.getCurrencyPair(this.state);
+
+    await onSubmitCallback({
+      sourceCurrency,
+      destCurrency,
+      amount: this.state.sourceAmount,
+    });
+
+    this.setState({ isHandlingSubmit: false });
+  }
+
+  tryFillDefaultCurrencies = (prevState: State) => {
+    if (prevState.sourceCurrency === null && this.props.defaultSource !== null) {
+      this.setState({ sourceCurrency: this.props.defaultSource });
+    }
+
+    if (prevState.destCurrency === null && this.props.defaultDest !== null) {
+      this.setState({ destCurrency: this.props.defaultDest });
+    }
+  }
+
+  getCurrencyPair = (state: State): [string, string] => {
+    return [
+      get(state, 'sourceCurrency.symbol'),
+      get(state, 'destCurrency.symbol'),
+    ];
+  }
```

Comment:
1) Why pass `state` as an argument if you can just access it inside the function? 
2) nit: instead of using `get`, you can access nested values via optional chaining, it's less prone to typos and it's fresh and fancy :) 
3) Both `get` and optional chaining might technically return `undefined`, which should be reflected in the return type.
How about something like:
```
getCurrencyPair = (): [?string, ?string] => {
  const { sourceCurrency, destCurrency } = this.state;
  return [ sourceCurrency?.symbol, destCurrency?.symbol ];
}
```